pkg/app/AppService.go

```diff
@@ -248,20 +248,22 @@ func (impl AppServiceImpl) UpdateApplicationStatusAndCheckIsHealthy(app v1alpha1
 	isHealthy := false
 	repoUrl := app.Spec.Source.RepoURL
 	// backward compatibility for updating application status - if unable to find app check it in charts
-	chart, err := impl.chartRepository.FindByGitRepoUrl(repoUrl)
+	charts, err := impl.chartRepository.FindChartsByGitRepoUrl(repoUrl)
 	if err != nil {
-		impl.logger.Errorw("error in fetching chart", "err", err, "chart", chart.ChartName)
+		impl.logger.Errorw("error in fetching chart", "err", err)
 		return isHealthy, err
 	}
+	if charts == nil || len(charts) == 0 {
+		impl.logger.Errorw("error in fetching chart", "err", "no charts found for git repo url")
+		return isHealthy, fmt.Errorf("no charts found for git repo url")
+	}
+	chart := charts[0]
```

Comment:
we are using first chart here. check this logic. 
and if this logic is correct, then return one entry from DB (limit 1)